### PR TITLE
feat: Add example Go stack

### PR DESCRIPTION
**Pull request by lym953**
### What does this PR do?
Adds an example Go CDK stack to demonstrate how to integrate Datadog with CDK.

I did this by:
1. create a demo CDK project following [Tutorial: Create your first AWS CDK app](https://docs.aws.amazon.com/cdk/v2/guide/hello_world.html)
2. add Datadog-related setup in `app.go`
3. modify `README.md`. Most of the content was copied from other example stacks.

#### Steps
1. run `go get` to get dependencies
2. run `go build` to build the binary file `app`
3. run `aws-vault exec sso-serverless-sandbox-account-admin -- cdk bootstrap --toolkit-stack-name yiming-go-cdk-test --qualifier yiming` to bootstrap the environment (optional)
4. run `aws-vault exec sso-serverless-sandbox-account-admin -- cdk deploy` to deploy the stack
5. in AWS UI, run the Lambda function a few times

#### Result
The Lambda function showed up on Datadog UI
image

### Additional Notes

[Feature Request: Datadog CDK Construct for Go](https://datadoghq.atlassian.net/browse/FRSLES-665)

Next steps:
1. Think how future updates will affect the `.tgz` file and whether it will break things
2. Maybe release a new version and write a release note. Right now the version is shown as `v0.0.0-20240801201308-5bce51ed3d25` on [pkg.go.dev](https://pkg.go.dev/github.com/DataDog/datadog-cdk-constructs/dist/go/ddcdkconstruct)
3. Understand why `npx projen` and `npx projen build` generate different content for `package.json`

### Types of Changes

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [x] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [x] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [ ] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -106,58 +143,63 @@ datadog.addForwarderToNonLambdaLogGroups([<LOG_GROUPS>])
...
+If the automatic implementation doesn't work for your case, please follow one of the two guides below.
 
 **Note: these alternate guides only work for Typescript.**
+
 <details>
   <summary>datadog-cdk version satisfied, but Datadog Github Integration NOT installed</summary>
 
-  If the Datadog Github Integration is not installed, you need to import the `datadog-ci` package and manually upload your Git metadata to Datadog.
-  We recommend you do this where your CDK Stack is initialized.
+If the Datadog Github Integration is not installed, you need to import the `datadog-ci` package and manually upload your Git metadata to Datadog.
```
Comment (urseberry) [inline README.md:154]:
```suggestion
If the Datadog Github integration is not installed, you need to import the `datadog-ci` package and manually upload your Git metadata to Datadog.
```

File: README.md [hunk 2]
```diff
@@ -106,58 +143,63 @@ datadog.addForwarderToNonLambdaLogGroups([<LOG_GROUPS>])
...
+We recommend you do this where your CDK Stack is initialized.
```
Comment (urseberry) [inline README.md:155]:
```suggestion
For the best results, import the `datadog-ci` package where your CDK Stack is initialized.
```
Feel free to replace "For the best results" with a description of the benefits of having the package and CDK Stack in the same place.

File: examples/go-stack/README.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Datadog CDK Go Example
+
+Use this example Go stack to try out the [datadog-cdk-constructs](https://github.com/DataDog/datadog-cdk-constructs) v2 library. It contains Node, Python and Go Lambda functions.
```
Comment (urseberry) [inline examples/go-stack/README.md:3]:
```suggestion
Use this example Go stack to try out the [datadog-cdk-constructs](https://github.com/DataDog/datadog-cdk-constructs) v2 library. It contains Node, Python, and Go Lambda functions.
```

File: examples/go-stack/README.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+## Getting Started
+
+1. Get a Datadog API key to be used to send monitoring data ([Datadog API keys documentation](https://docs.datadoghq.com/account_management/api-app-keys/#add-an-api-key-or-client-token)).
```
Comment (urseberry) [inline examples/go-stack/README.md:7]:
```suggestion
1. Get a Datadog API key to send monitoring data ([Datadog API keys documentation](https://docs.datadoghq.com/account_management/api-app-keys/#add-an-api-key-or-client-token)).
```

**Comment by lym953** [comment]
I'm going to move the package to a separate repo `datadog-cdk-constructs-go`, so the demo stack will need to be updated. Closing this PR for now.

**Pull request closed without merging**